src/__tests__/posthog-core.ts

```diff
@@ -337,6 +337,14 @@ describe('posthog core', () => {
         })
     })
 
+    describe('page', () => {
+        it('has the correct url', () => {
+            const testUrl = 'https://example.com'
+            const captureData = posthogWith(defaultConfig, defaultOverrides).page(testUrl, {}, {})
```

Comment:
elsewhere we test that after capture particular values have been passed to _send_request

i.e. capture ran as expected.
here we're testing the return value from capture 

so we should either

* test that properties are set / passed through and end up being passed out of capture
* or simply mock capture and test that `page` calls capture (and so only needs to rely on capture being correct)

that second lets us test that 

* properties are passed through
* options are passed through
* url is set correctly

it'd make the contract of page much clearer in the tests as well

---

why? i could reimplement page without calling capture but returning the expected data directly and the tests would pass but the library would be broken :)

Reply:
would be good to test `(url, {}, {})`, `(url, {some: 'prop')`, `(url)` (or some similar combination of possible inputs to clarify the contract

Reply:
@pauldambra This makes sense. To paraphrase, `page` is tightly coupled to the functionality of `capture` and the test cases should reflect that, so that when `page` is accidentally rewritten as another implementation without calling `capture` (decoupled from `capture`), the test cases should fail, right?

Reply:
yep exactly... all `page` should do is maybe transform the args passed to it and call capture with them

so we only need to test that it calls capture as expected with some example inputs

that way the tests correctly describe and protect the contract